ui/src/components/CheckboxTree/validation.ts

```diff
@@ -0,0 +1,26 @@
+import Validator from '../../util/Validator';
+import { Row } from './types';
+
+type MaybeError =
+    | {
+          errorField: string;
+          errorMsg: string;
+      }
+    | false;
+
+/**
+ * Validate the required field has value
+ * @param {string} field
+ * @param {string} label
+ * @param {object} [rows]
+ * @returns {Error|false}
+ */
+
+export function checkValidationForRequired(field: string, label: string, rows: Row[]): MaybeError {
+    let errorMessage: MaybeError = false;
+    if (!rows.some((row) => row.checkbox?.defaultValue === true)) {
```

Comment:
not sure why we check defaultValue here, should it be about current value not default one?

I might be misunderstanding it but pls add some comment then why we check it this way.

Reply:
refactored